docs/development/best-practices.md

```diff
@@ -157,11 +157,11 @@ Avoid refactoring the code and tests at the same time, this can mask regression
 For `WARN`, `ERROR` and `FATAL` log messages use logger metadata.
 Also use logger metadata the result is a complex metadata object needing a multiple-line pretty stringification.
 
-For `INFO` log messages inline the metadata into the log message.
+For `INFO`, `DEBUG` and `TRACE` log messages inline the metadata into the log message.
```

Comment:
```suggestion
For `INFO` and `DEBUG` log messages inline the metadata into the log message where feasible.
```

It's rare to look at TRACE messages so we don't need to care how they are structured